src/framework/audio/thirdparty/lame/CMakeLists.txt

```diff
@@ -72,7 +72,9 @@ target_no_warning(${MODULE} -Wno-implicit-fallthrough)
 target_no_warning(${MODULE} -Wno-empty-body)
 target_no_warning(${MODULE} -Wno-attributes)
 target_no_warning(${MODULE} -Wno-absolute-value) # we need this for C code in this module , but not for its C++ code
-#target_no_warning(${MODULE} -Wno-tautological-pointer-compare)
+target_no_warning(${MODULE} -Wno-tautological-pointer-compare) # we need this for C code in this module , but not for its C++ code
```

Comment:
When using `target_compile_options` directly, it is possible to do
```cmake
target_compile_options(${MODULE} PRIVATE $<$<COMPILE_LANGUAGE:C>:-Wno-tautological-pointer-compare>
```
This `$<$<…>…>` phenomenon is called generator expressions, see this one: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:COMPILE_LANGUAGE

To be honest, I'm not the biggest fan of `target_no_warning`; it seems an unnecessary wrapper around simply 
```cmake 
if (CC_IS_GCC OR CC_IS_CLANG)
    target_compile_options(…)
endif()
```

Reply:
On second thoughts, perhaps `target_no_warning` is kind of fine because overall it does reduce duplication; but then for this generator expression I think you'd have to introduce some "special case" like `-Wonly-c-no-tautological-pointer-compare`

Reply:
Now I'm confused. What exactly should this get changed to?

Reply:
In SetupCompileWarnings.cmake, replace
```
    elseif(WNAME STREQUAL "-Wno-tautological-pointer-compare")
```
with
```
    elseif(WNAME STREQUAL "-Wc-no-tautological-pointer-compare")
        set(GCC_Warning $<$<COMPILE_LANGUAGE:C>:-Wno-tautological-pointer-compare>)
        unset(CLANG_Warning) # or set Clang_Warning to the same, iff Clang also warns about this
```

Here, add
```
target_no_warning(${MODULE} -Wc-no-tautological-pointer-compare)
```
Same for all other warnings that apply to C only

Reply:
Ah!

Reply:
There are still these 3:
```
2024-11-27T18:01:29.0713372Z [164/1322] Building CXX object src/framework/audio/opusenc/opus/CMakeFiles/opus.dir/opus_autogen/mocs_compilation.cpp.o
2024-11-27T18:01:29.0715828Z cc1plus: warning: command-line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
2024-11-27T18:01:29.0717373Z cc1plus: warning: command-line option ‘-Wnested-externs’ is valid for C/ObjC but not for C++
2024-11-27T18:01:29.0719048Z cc1plus: note: unrecognized command-line option ‘-Wno-truncate’ may have been intended to silence earlier diagnostics
```
Maybe you have a clever idea how to get rid of those?

Reply:
It looks like the first two are set in Opus's own files, so we have basically no control over that. But the problem seems that the Opus files are not expecting to compile any cpp files. That `mocs_compilation.cpp` file is generated by CMake, but that's not necessary for non-Qt modules of course. I'll create a separate PR for that.

For `-Wno-truncate`: it looks like that option just doesn't exist and I can't find any evidence online that it has ever existed. That one does come from `target_no_warning`, so you can just remove it.

Reply:
I don't see where that stems from
in SetupCompileWarnings.cmake is apparently mapped to some MSVC options
```
    elseif(WNAME STREQUAL "-Wno-truncate")
        set(MSVC_Warning /wd4310 /wd4311)
```

Reply:
Found it

Reply:
Can't remove it there though, MSVC needs it
So I'll unset it for GCC in SetupCompileWarnings.cmake